clang/test/SemaSYCL/sycl-restrict.cpp

```diff
@@ -183,7 +187,9 @@ __attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
   kernelFunc();
   a_type ab;
   a_type *p;
-  // expected-note@+1 {{called by 'kernel_single_task'}}
+  // expected-error@+1 {{__float128 is not supported on this target}}
+  __float128 A;
+  // expected-note@+1 4{{called by 'kernel_single_task'}}
```

Comment:
The picture is changed a bit after the patch from Erich, but repeating notes happens because any deferred diagnostic is emitted with call stack notes and if call stack merges for several diagnostics we will see several same notes.